lifecycle/index.js

```diff
@@ -0,0 +1,76 @@
+const getOrCreate = (dest, key, getValue) => {
+  if (!dest.has(key)) dest.set(key, getValue())
+  return dest.get(key)
+}
+
+export const container = new Map()
+
+const on = (store, pluginHandler, eventKey, eventHandler) => {
+  let storeContainer = getOrCreate(container, store, () => new Map())
+  getOrCreate(storeContainer, `${eventKey}-d`, () =>
+    eventHandler(store, (original, methods) => {
+      storeContainer.get(eventKey).reduceRight((shared, h) => {
+        h({ original, shared, ...methods })
+        return shared
+      }, {})
+    })
+  )
+  getOrCreate(storeContainer, eventKey, () => []).push(pluginHandler)
+  return () => {
+    let current = storeContainer.get(eventKey)
+    let index = current.indexOf(pluginHandler)
+    current.splice(index, 1)
+    if (!current.length) {
+      storeContainer.delete(eventKey)
+      storeContainer.get(`${eventKey}-d`)()
+      storeContainer.delete(`${eventKey}-d`)
+    }
+    if (!storeContainer.size) container.delete(store)
+  }
+}
+
+export const onCreate = (destStore, cb) =>
```

Comment:
Can we rename it to `onStart` (to rhythm with `onStop`)? We may need `onCreate` to run listener on store creating from store template.